src/main/java/net/joshka/junit/json/params/JsonArgumentsProvider.java

```diff
@@ -38,16 +42,24 @@ public Stream<? extends Arguments> provideArguments(ExtensionContext extensionCo
 
     private Stream<? extends Arguments> getArguments(String value) throws IOException {
         try (Reader reader = new StringReader(value)) {
-            return values(reader).map(Arguments::of);
+            if (this.targetClass.equals(Object.class)) {
+                return values(reader, this.isArrayClass)
+                        .map(Arguments::of);
+            }
+            return values(reader, this.isArrayClass)
+                    .map(json -> JsonConverter.convert(json, this.targetClass))
+                    .map(Arguments::of);
         }
     }
 
-    private static Stream<JsonValue> values(Reader reader) {
+    private static Stream<JsonValue> values(Reader reader, boolean arrayClass) {
         try (JsonReader jsonReader = Json.createReader(reader)) {
             JsonStructure structure = jsonReader.read();
-            return structure.getValueType() == JsonValue.ValueType.ARRAY
-                    ? structure.asJsonArray().stream()
-                    : Stream.of(structure);
+            Stream<JsonValue> values = Stream.of(structure);
+            if (structure.getValueType() == JsonValue.ValueType.ARRAY) {
+                values = arrayClass ? values : structure.asJsonArray().stream();
+            }
+            return values;
```

Comment:
I think this is possibly simplified as:
```java
bool isArray = (structure.getValueType() == JsonValue.ValueType.ARRAY);
return (isArray && !this.isArrayClass)
  ? structure.asJsonArray().stream
  : Stream.of(structure);
```
Does that make sense? That avoids calling Stream.of unless it's needed. It's unlikely this is a perf issue, but I think it's clearer. Thoughts?